code/graphics/generic.cpp

```diff
@@ -219,6 +224,29 @@ int generic_anim_stream(generic_anim *ga)
 
 		ga->previous_frame = -1;
 	}
+	else if (ga->type == BM_TYPE_PNG) {
+		if (ga->png.anim == nullptr) {
+			try {
+				ga->png.anim = new apng::apng_ani(ga->filename);
```

Comment:
I hadn't thought of that. I checked the code and it should be releasing the pointer properly so if it turns out to be too much work then the current state is fine.